pyproject.toml

```
@@ -10,9 +10,11 @@ repository = "https://github.com/J-Quants/jquants-api-client-python"
 keywords = ["jquants", "api", "client", "J-Quants"]
 
 [tool.poetry.dependencies]
-python = "^3.8"
-pandas = "^1.4.3"
+python = ">=3.7.1,<4.0"
 requests = "^2.28.1"
+types-requests = "^2.28.5"
+types-python-dateutil = "^2.8.19"
+pandas = ">=1.2.0"
```

https://numpy.org/devdocs/release/1.22.0-notes.html
numpy は 1.22.0 から Python 3.7 サポートを落としていて、Python 3.7 がサポートされている最後のバージョンは 1.21.6 のようです。

試していなくてすみませんが、以下のような指定ができたりしますでしょうか？

```suggestion
pandas = ">=1.2.0"
numpy = {version = "^1.21.6", python = ">=3.7.1,<3.8"}
```